cmd/goal/clerk.go

```diff
@@ -918,32 +922,12 @@ var splitCmd = &cobra.Command{
 	Long:  `Split a file containing many transactions.  The input file must contain one or more transactions.  These transactions will be written to individual files.`,
 	Args:  validateNoPosArgsFn,
 	Run: func(cmd *cobra.Command, args []string) {
-		data, err := readFile(txFilename)
-		if err != nil {
-			reportErrorf(fileReadError, txFilename, err)
-		}
-
-		dec := protocol.NewMsgpDecoderBytes(data)
-
-		var txns []transactions.SignedTxn
-		for {
-			var txn transactions.SignedTxn
-			err = dec.Decode(&txn)
-			if err == io.EOF {
-				break
-			}
-			if err != nil {
-				reportErrorf(txDecodeError, txFilename, err)
-			}
-
-			txns = append(txns, txn)
-		}
-
+		txns := decodeTxnsFromFile(txFilename)
 		outExt := filepath.Ext(outFilename)
 		outBase := outFilename[:len(outFilename)-len(outExt)]
 		for idx, txn := range txns {
 			fn := fmt.Sprintf("%s-%d%s", outBase, idx, outExt)
-			err = writeFile(fn, protocol.Encode(&txn), 0600)
+			err := writeFile(fn, protocol.Encode(&txn), 0600)
```

Comment:
I think the reviewdog is suggesting this should be `protocol.Encode(&txns[idx])` but since it's writing to the file I'm not sure it matters.

Reply:
I think the reviewdog is suggesting something like https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop, and I found it confusing in some earlier PRs, e.g., https://github.com/algorand/go-algorand/pull/5059#discussion_r1088504367

Reply:
It doesn't matter in this case, but since I touched that line reviewdog is now complaining about it. I'll fix in a commit soon